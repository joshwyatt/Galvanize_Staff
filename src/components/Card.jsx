import React from 'react';

import '../../style/card.css';
export const Card = (props) => {
  const { name, bio, picture, role, link } = props;
  return (
    <div className="col s12 m6 l4">
      <div className="card">
        <div className="card-image waves-effect waves-block waves-light" style={style.imageContainer}>
          <img className="activator" src={picture} style={style.image}/>
        </div>
        <div className="card-content">
          <span className="card-title activator grey-text text-darken-4">{name}<i className="material-icons right">more_vert</i></span>
          <p><a target="_blank" href={link}>{role}</a></p>
        </div>
        <div className="card-reveal">
          <span className="card-title grey-text text-darken-4">Bio<i className="material-icons right">close</i></span>
          <p>{bio}</p>
        </div>
      </div>
    </div>
  );
};

const style = {
  imageContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    height: 300,
    flex: 1,
    width: null,
  },
};


export default Card;