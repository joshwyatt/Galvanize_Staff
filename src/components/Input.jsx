import React from 'react';
import { connect } from 'react-redux';
import { updateQuery } from '../actions';

export const Input = (props) => {
  const { query, updateQuery } = props;
  const logChange = (e) => {
    updateQuery(e.target.value);
  };

  return (
    <div>
      <input type="text" autoFocus={true} onChange={logChange} value={query}/>
    </div>
  );
};


const mapStateToProps = ({ query }) => {
  return {
    query,
  };
};

export default connect(mapStateToProps, { updateQuery })(Input);