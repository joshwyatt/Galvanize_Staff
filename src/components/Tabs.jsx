import React from 'react';
import { connect } from 'react-redux';
import { updateTag } from '../actions';
import '../../style/tabs.css';

export const Tabs = props => {
  const { updateTag } = props;

  return (
    <div className="row">
      <div className="col s12">
        <ul className="tabs">
          <li className="tab col s2"><a className="active" href="#hot" onClick={() => updateTag()}>All</a></li>
          <li className="tab col s2"><a href="#New" onClick={() => updateTag('DSI')}>DSI</a></li>
          <li className="tab col s2"><a href="#Rising" onClick={() => updateTag('WDI')}>WDI</a></li>
        </ul>
      </div>
    </div>
  );
};

const style = {
  textStyle: {
    color: 'orange',
  }
}

export default connect(null, { updateTag })(Tabs);

