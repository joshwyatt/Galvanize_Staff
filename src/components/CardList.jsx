import React from 'react';
import { connect } from 'react-redux';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
import '../../style/transition.css';
import SelectedStaffSelector from '../selectors/selected_staff';

import Card from './Card';

export const CardList = props => {

  const generateList = (list=[]) => {
    return list.map((member, key) => {
      return (
        <Card {...member} key={key}/>
      );
    });
  };
  return (
    <div className="row">
      <CSSTransitionGroup
        transitionName="example"
        transitionAppear={true}
        transitionLeaveTimeout={1000}
        transitionEnterTimeout={1000}
        transitionAppearTimeout={1000}
        transitionEnter={true}
        transitionLeave={true}>
        {generateList(props.staff)}
      </CSSTransitionGroup>
    </div>
  );
};


const mapStateToProps = state => {
  return {
    staff: SelectedStaffSelector(state)
  };
};

//CardList.PropTypes = {
//  staff: [],
//  tabStaff: [],
//};

export default connect(mapStateToProps)(CardList);
