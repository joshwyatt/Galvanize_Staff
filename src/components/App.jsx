import React from 'react';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
import '../../style/transition.css';

import CardList from './card/CardList';
import Input from './Input';
import Tabs from './Tabs';

const App = (props) => {

  return (
    <div className="container">
      <h2 className="center">Galvanize Staff</h2>
      <div className="container">
        <Input/>
      </div>
      <Tabs/>
      <CSSTransitionGroup
        transitionName="example"
        transitionAppear={true}
        transitionAppearTimeout={1000}
        transitionEnter={false}
        transitionLeave={false}>
        <CardList/>
      </CSSTransitionGroup>
    </div>
  );
};


export default App;
