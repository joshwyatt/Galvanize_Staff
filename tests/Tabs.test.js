import { shallow } from 'enzyme';
import React from 'react';
import { Tabs } from '../src/components/Tabs';


describe('Tabs', () => {
  it('Welcome renders its child components', () => {
    const tabs = shallow(<Tabs />);
    expect(tabs).toMatchSnapshot();
  });
});