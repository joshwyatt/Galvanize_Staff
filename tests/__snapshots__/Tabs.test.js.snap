exports[`Tabs Welcome renders its child components 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="row">
    <div
        className="col s12">
        <ul
            className="tabs">
            <li
                className="tab col s2">
                <a
                    className="active"
                    href="#hot"
                    onClick={[Function]}>
                    All
                </a>
            </li>
            <li
                className="tab col s2">
                <a
                    href="#New"
                    onClick={[Function]}>
                    DSI
                </a>
            </li>
            <li
                className="tab col s2">
                <a
                    href="#Rising"
                    onClick={[Function]}>
                    WDI
                </a>
            </li>
        </ul>
    </div>
</div>,
  "nodes": Array [
    <div
      className="row">
      <div
            className="col s12">
            <ul
                  className="tabs">
                  <li
                        className="tab col s2">
                        <a
                              className="active"
                              href="#hot"
                              onClick={[Function]}>
                              All
                        </a>
                  </li>
                  <li
                        className="tab col s2">
                        <a
                              href="#New"
                              onClick={[Function]}>
                              DSI
                        </a>
                  </li>
                  <li
                        className="tab col s2">
                        <a
                              href="#Rising"
                              onClick={[Function]}>
                              WDI
                        </a>
                  </li>
            </ul>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="row">
          <div
                    className="col s12">
                    <ul
                              className="tabs">
                              <li
                                        className="tab col s2">
                                        <a
                                                  className="active"
                                                  href="#hot"
                                                  onClick={[Function]}>
                                                  All
                                        </a>
                              </li>
                              <li
                                        className="tab col s2">
                                        <a
                                                  href="#New"
                                                  onClick={[Function]}>
                                                  DSI
                                        </a>
                              </li>
                              <li
                                        className="tab col s2">
                                        <a
                                                  href="#Rising"
                                                  onClick={[Function]}>
                                                  WDI
                                        </a>
                              </li>
                    </ul>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="row">
          <div
                    className="col s12">
                    <ul
                              className="tabs">
                              <li
                                        className="tab col s2">
                                        <a
                                                  className="active"
                                                  href="#hot"
                                                  onClick={[Function]}>
                                                  All
                                        </a>
                              </li>
                              <li
                                        className="tab col s2">
                                        <a
                                                  href="#New"
                                                  onClick={[Function]}>
                                                  DSI
                                        </a>
                              </li>
                              <li
                                        className="tab col s2">
                                        <a
                                                  href="#Rising"
                                                  onClick={[Function]}>
                                                  WDI
                                        </a>
                              </li>
                    </ul>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
